grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -343,15 +448,21 @@ public GrpcServiceBuilder useClientTimeoutHeader(boolean useClientTimeoutHeader)
     public GrpcService build() {
         final HandlerRegistry handlerRegistry;
         if (USE_COROUTINE_CONTEXT_INTERCEPTOR) {
-            final HandlerRegistry registry = registryBuilder.build();
             final ServerInterceptor coroutineContextInterceptor =
                     new ArmeriaCoroutineContextInterceptor(useBlockingTaskExecutor);
-            final HandlerRegistry.Builder registryBuilder = new HandlerRegistry.Builder();
-            for (ServerServiceDefinition serviceDefinition : registry.services()) {
-                registryBuilder.addService(ServerInterceptors.intercept(serviceDefinition,
-                                                                        coroutineContextInterceptor));
+            final HandlerRegistry.Builder newRegistryBuilder = new HandlerRegistry.Builder();
+
+            for (Entry entry : registryBuilder.entries()) {
+                final MethodDescriptor<?, ?> methodDescriptor = entry.method();
+                final ServerServiceDefinition intercepted =
+                        ServerInterceptors.intercept(entry.service(), coroutineContextInterceptor);
+                if (methodDescriptor != null) {
+                    newRegistryBuilder.addService(entry.path(), intercepted, methodDescriptor);
```

Comment:
nit: could remove if and just call `newRegistryBuilder.addService(entry.path(), intercepted, entry.method());` for simplicity. (We can make the method take nullable `methodDecriptor` because it's hidden API)